File: clients/go/examples/tables-create-index-map-values.go
package main

import (
	"context"
	"log"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()

	// Get an existing table
	client := astra.NewClient()

	database := client.Database(
		"**API_ENDPOINT**",
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	table := database.Table("**TABLE_NAME**")

	// Index a the values of a map column
	err := table.CreateIndex(
		ctx,
		"**INDEX_NAME**",
		map[string]string{"**MAP_COLUMN_NAME**": "$values"},
	)
	if err != nil {
		log.Fatal(err)
	}
}

File: clients/typescript/examples/tables-create-index-map-values.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing table
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const table = database.table("**TABLE_NAME**");

// Index a column
(async function () {
  await table.createIndex("**INDEX_NAME**", {
    "**MAP_COLUMN_NAME**": "$values",
  });
})();
